optional_components/embeddings/app/main.py

```diff
@@ -0,0 +1,78 @@
+from typing import List
+
+from config import API_KEY, HUGGINGFACE_MODEL
+from fastapi import Depends, FastAPI, HTTPException, status
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from pydantic import BaseModel
+from torch import Tensor
+from transformers import AutoConfig, AutoModel, AutoTokenizer
+
+app = FastAPI()
+
+security = HTTPBearer()
+
+
+@app.on_event("startup")
+def load_model():
+    """Load the model and tokenizer on startup"""
+    global tokenizer
+    global model
+    tokenizer = AutoTokenizer.from_pretrained(HUGGINGFACE_MODEL)
+    config = AutoConfig.from_pretrained(HUGGINGFACE_MODEL)
+    model = AutoModel.from_pretrained(HUGGINGFACE_MODEL, config=config)
```

Comment:
Is there a way to only run this if we haven't mounted the files?
Generally do we want to download always from huggingface hub or allow users to mount a volume containing their own embeddings?

We can create a ticket and discuss this in the future!

Reply:
Yes there are several approaches to this, I used this approach because it is easy to build on docker. Next steps would be:
- Update the Dockerfile to download a model from anywhere 
- Copy to the environment
- Use the local model when deployed.